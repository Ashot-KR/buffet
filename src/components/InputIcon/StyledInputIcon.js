import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import colors from '../../assets/styles/colors';
import sizes from '../../assets/styles/sizes';

const StyledIcon = styled.span`
  position: absolute;
  top: 0;
  width: ${sizes.input.height};
  height: ${sizes.input.height};
  display: flex;

  /*
   * Input with left icon
   */
  ${props =>
    props.type !== 'text' &&
    css`
      background-color: ${colors.greyIconBkgd};
      left: 0;
      ::before {
        font-family: 'FontAwesome';
        font-weight: 600;
        font-size: 14px;
        line-height: 100%;
        display: inline-block;
        vertical-align: middle;
        margin: auto;
        color: ${colors.greyIconColor};
      }
    `}}
    
  /*
   * Specific icon
   */

  // Search
  ${props =>
    props.type === 'search' &&
    css`
      ::before {
        content: '\f002';
      }
    `}}

  // Email
  ${props =>
    props.type === 'email' &&
    css`
      ::before {
        content: '\f1fa';
      }
    `}}
    
  // Password
  ${props =>
    props.type === 'password' &&
    css`
      background-color: transparent;
      ::before {
        content: '\f06e';
      }
    `}}
       
`;

StyledIcon.defaultProps = {
  type: 'search',
};
StyledIcon.propTypes = {
  type: PropTypes.string,
};

export default StyledIcon;
