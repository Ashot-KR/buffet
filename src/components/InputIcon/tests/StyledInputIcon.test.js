import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import StyledInputIcon from '../index';

let renderedComponent;
const renderComponent = (props = {}) => mount(<StyledInputIcon {...props} />);

describe('<StyledInputIcon />', () => {
  describe('<StyledInputIcon /> attributes', () => {
    afterEach(() => {
      renderedComponent.unmount();
    });
    it('should have a default type equals to search', () => {
      renderedComponent = renderComponent({ type: 'search' });
      const defaultType = 'search';
      expect(renderedComponent.at(0).prop('type')).toBe(defaultType);
    });
  });
  describe('<StyledInputIcon /> icon attributes', () => {
    it('should render the search icon', () => {
      const tree = renderer.create(<StyledInputIcon type="search" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the email icon', () => {
      const tree = renderer.create(<StyledInputIcon type="email" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the password icon', () => {
      const tree = renderer
        .create(<StyledInputIcon type="password" />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
