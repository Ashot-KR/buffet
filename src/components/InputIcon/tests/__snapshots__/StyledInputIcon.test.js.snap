// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledInputIcon /> <StyledInputIcon /> icon attributes should render the email icon 1`] = `
<span
  className="sc-bdVaJa lkPXav"
  type="email"
/>
`;

exports[`<StyledInputIcon /> <StyledInputIcon /> icon attributes should render the password icon 1`] = `
<span
  className="sc-bdVaJa fsSfwc"
  type="password"
/>
`;

exports[`<StyledInputIcon /> <StyledInputIcon /> icon attributes should render the search icon 1`] = `
<span
  className="sc-bdVaJa otneW"
  type="search"
/>
`;
